### DISCUSSION TITLE
Monarch: Parentheses are grouped as a single token

### DISCUSSION THREAD

**Discussion by vmlopezr** [microsoft/monaco-editor#3673] [Q&A]
Hello, i'm trying to write some logic for custom auto-complete but am having issues with the tokenizer.  Whenever adjacent parantheses are encountered, they are grouped as one token.

This causes issues when cases such as 'function(function2())' come up. Is there a way to separate each parenthesis character as separate tokens? 

**monaco-editor version:** 0.30.1
**Browser:** Firefox, Chrome, Safari
**OS:** MacOS
**Playground code that reproduces the issue:**
See:
https://microsoft.github.io/monaco-editor/monarch.html

![image](https://user-images.githubusercontent.com/35988895/143491612-7ab653ed-73ef-460f-b629-5cc857951e55.png)

- Press F1 and select 'Developer: Inspect Tokens'. 
- Move the cursor to the parentheses at 'main()'. See that '()' is listed as a single token.

**Expected:**
- Each '(', ')' should be a individual token.

**Comment by rcjsuen** [comment]
Unrelated but looks like there's a totally different where the text's X offset is off as the `())` starts outside the hover... 🤔
